### ISSUE TITLE
ejabberdctl export2sql fails

### ISSUE THREAD

**Issue by amitojc** [processone/ejabberd#1638]
> What version of ejabberd are you using?

17.01

> What operating system (version) are you using?

Arch Linux

> How did you install ejabberd (source, package, distribution)?

Package

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

As mentioned in #1118 and #1166 , I added option db_type: sql to all these modules before running the export2sql command
```
mod_announce
mod_caps
mod_irc
mod_last
mod_muc
mod_offline
mod_privacy
mod_private
mod_pubsub
mod_roster
mod_shared_roster
mod_vcard
mod_vcard_xupdate
```

However, the command still fails with error in mod_pubsub. 
```
[root@localhost ejabberd]# ejabberdctl export2sql example.com /tmp/ejab.sql
Problem 'error undef' occurred executing the command.
Stacktrace: [{mod_pubsub,export,[<<"example.com">>],[]},
             {ejd2sql,export,3,[{file,"src/ejd2sql.erl"},{line,87}]},
             {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
             {ejd2sql,export,2,[{file,"src/ejd2sql.erl"},{line,70}]},
             {ejabberd_ctl,call_command,4,
                           [{file,"src/ejabberd_ctl.erl"},{line,329}]},
             {ejabberd_ctl,try_call_command,4,
                           [{file,"src/ejabberd_ctl.erl"},{line,298}]},
             {ejabberd_ctl,process2,4,
                           [{file,"src/ejabberd_ctl.erl"},{line,247}]},
             {ejabberd_ctl,process,2,
                           [{file,"src/ejabberd_ctl.erl"},{line,225}]}]
```

I got this error even when I hadn't even enabled mod_pubsub in ejabberd.yml. So just to be sure, I enabled mod_pubsub, and specified db_type: sql for mod_pubsub. But that doesn't solve the problem either. 

Any help will be greatly appreciated.

**Comment by zinid** [comment]
I believe this has been already fixed in 17.03.

**Comment by badlop** [comment]
Right, mod_pubsub can't export mnesia data to SQL (#1571), so it shouldn't attempt to export it (#1584).
